client/src/components/ExpandedRow.tsx

```diff
@@ -0,0 +1,62 @@
+import type {
+    ExpandedRowProps,
+    UpdateEventProps,
+} from '../lib/types';
+import React from 'react';
+import {useState} from 'react';
+import ExpandedRowDataView from './ExpandedRowDataView';
+import ExpandedRowDataEdit from './ExpandedRowDataEdit';
+
+const ExpandedRow = ({data, onUpdate, user}: ExpandedRowProps) => {
+    const [errorMessage, setErrorMessage] = useState('');
+    const [editing, setEditing] = useState(false);
+
+    const handleUpdate = (props: UpdateEventProps) => {
+        const {data, action} = props;
+        setErrorMessage('');
+        switch (action) {
+            case 'save':
+                setEditing(false);
+                onUpdate(props);
+                break;
+            case 'edit':
+                setEditing(true);
+                break;
+            case 'cancel':
+                setEditing(false);
+                break;
+            default:
+                setEditing(false);
+                onUpdate(props);
+                break;
+        }
+    };
+
+    const handleError = (message: string) => {
+        setErrorMessage(message);
+    };
+
+    return (
+        <div style={{paddingLeft: '3em'}} className="expanded-row">
+            {!editing && (
+                <ExpandedRowDataView
+                    data={data}
+                    onError={handleError}
+                    onUpdate={handleUpdate}
+                    user={user}
+                />
+            )}
+            {editing && (
+                <ExpandedRowDataEdit
+                    data={data}
+                    onError={handleError}
+                    onUpdate={handleUpdate}
+                    user={user}
+                />
+            )}
```

Comment:
Not a blocker, but would it be worth passing `isEditting` as a prop to a single `ExpandableRow` component that can render either the edit or view component version?

Reply:
It was Pattie's suggestion to keep the View vs Editing component separate to keep the code smaller;  This also allow us to focus on editing view separate from regular view.  They are two separate control.

Reply:
Ah, this component is the `ExpandedRow` component haha